### ISSUE TITLE
MockFileSystem doesn't throw UnauthorizedAccessException when overriding a hidden file

### ISSUE THREAD

**Issue by robertmuehsig** [tathamoddie/System.IO.Abstractions#92]
The MockFileSystem doesn't behave the same way as the real FileSystem. 
I know, its a mock, but this would be handy ;)

Testcode (if you want a PR with a real test case, just tell me)

            // Arrange
            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
            {
                { @"C:\\Document\\Stuff\\MyDoc.foobar", new MockFileData("this is hidden") },
            });
            fileSystem.File.SetAttributes(@"C:\\Document\\Stuff\\MyDoc.foobar", FileAttributes.Hidden);

            // Act - this should thrown an UnauthorizeAccessException 
            fileSystem.File.WriteAllText(@"C:\\Document\\Stuff\\MyDoc.foobar", "hello world");

            // No Exception thrown, but it should - like in the real world 
            // http://stackoverflow.com/questions/2246990/how-do-i-write-to-a-hidden-file
            var content = fileSystem.File.ReadAllText(@"C:\\Document\\Stuff\\MyDoc.foobar");

content contains "hello world".

**Comment by manne** [comment]
hi @robertmuehsig
 i can not reproduce the behavior of the last call with the real file system.
````var content = fileSystem.File.ReadAllText(@"C:\\Document\\Stuff\\MyDoc.foobar");````

Following code is used:
````
    var p = @"c:\tmp\test12.txt";
    File.SetAttributes(p, FileAttributes.Hidden);   
    File.ReadAllText(p);
````
it does not throw an UnauthorizeAccessException exception with .net 4.0.

can you give me more details.

**Comment by robertmuehsig** [comment]
The UnauthorizedAccessException is only thrown when you try to override that file -  reading is fine.
________________________________
Von: Manuel Pfemeter<mailto:notifications@github.com>
Gesendet: ý24.ý01.ý2015 20:10
An: tathamoddie/System.IO.Abstractions<mailto:System.IO.Abstractions@noreply.github.com>
Cc: Robert Mühsig<mailto:robert.muehsig@code-developer.de>
Betreff: Re: [System.IO.Abstractions] MockFileSystem doesn't throw UnauthorizedAccessException when overriding a hidden file (#92)


hi @robertmuehsig<https://github.com/robertmuehsig>
i can not reproduce the behavior of the last call with the real file system.
var content = fileSystem.File.ReadAllText(@"C:\\Document\\Stuff\\MyDoc.foobar");

Following code is used:

    var p = @"c:\tmp\test12.txt";
    File.SetAttributes(p, FileAttributes.Hidden);
    File.ReadAllText(p);


it does not throw an UnauthorizeAccessException exception with .net 4.0.

can you give me more details.

**Comment by manne** [comment]
fixed with #99

**Comment by robertmuehsig** [comment]
:+1: Awesome!
